src/widgets/conditionalWidget/matrix2d.js

```diff
@@ -0,0 +1,144 @@
+const emptyObject = {};
+
+const isObjectEmpty = obj => (
+    !!obj && Object.keys(obj).length === 0
+);
+
+const emptyArray = [];
+
+const getDimensionOptions = ({ dimensions = emptyArray } = {}) => (
+    dimensions.map(r => ({
+        key: r.id,
+        title: r.title,
+    }))
+);
+
+const getSubdimensionOptions = ({ dimensions = emptyArray } = {}) => (
+    dimensions.reduce((acc, r) => [
+        ...((r || emptyObject).subdimensions || emptyArray).map(c => ({
+            key: c.id,
+            title: c.title,
+        })),
+        ...acc,
+    ], [])
+);
+
+const getSectorOptions = ({ sectors = emptyArray } = {}) => (
+    sectors.map(r => ({
+        key: r.id,
+        title: r.title,
+    }))
+);
+
+const getSubsectorOptions = ({ sectors = emptyArray } = {}) => (
+    sectors.reduce((acc, r) => [
+        ...((r || emptyObject).subsectors || emptyArray).map(c => ({
+            key: c.id,
+            title: c.title,
+        })),
+        ...acc,
+    ], [])
+);
+
+const containsDimension = {
+    title: 'Contains dimension',
+    attributes: [{
+        key: 'dimension',
+        type: 'select',
+        title: 'Dimension',
+        options: getDimensionOptions,
+        keySelector: d => d.key,
+        labelSelector: d => d.title,
+    }],
+    test: ({ value = {} }, { dimension }) => {
+        const dimensionValue = value[dimension];
+        if (!dimensionValue) {
+            return false;
+        }
+
+        const selectedKeys = Object.keys(dimensionValue)
+            .filter(k => !isObjectEmpty(dimensionValue[k]));
+        return selectedKeys.length !== 0;
+    },
+};
+
+const containsSubdimension = {
+    title: 'Contains subdimension',
+    attributes: [{
+        key: 'subdimension',
+        type: 'select',
+        title: 'Subdimension',
+        options: getSubdimensionOptions,
+        keySelector: d => d.key,
+        labelSelector: d => d.title,
+    }],
+    test: ({ value = {} }, { subdimension }) => {
+        const dimensionOfSubdimension = Object.keys(value)
+            .find(v => (value[v] || emptyObject)[subdimension] !== undefined);
+        if (!dimensionOfSubdimension) {
+            return false;
+        }
+        return !isObjectEmpty((value[dimensionOfSubdimension] || emptyObject)[subdimension]);
+    },
+};
+
+const containsSector = {
+    title: 'Contains sector',
+    attributes: [{
+        key: 'sector',
+        type: 'select',
+        title: 'Sector',
+        options: getSectorOptions,
+        keySelector: d => d.key,
+        labelSelector: d => d.title,
+    }],
+    test: ({ value = {} }, { sector }) => (
+        Object.keys(value).some((v) => {
+            const subdimen = value[v];
+            if (isObjectEmpty(subdimen)) {
```

Comment:
Returning true/false value can be avoided (as said by Dahal)

Reply:
Dahal suggests the code is cleaner this way.